server/fleet/windows_mdm.go

```diff
@@ -211,3 +212,42 @@ type MDMWindowsProfileContents struct {
 	SyncML   []byte `db:"syncml"`
 	Checksum []byte `db:"checksum"`
 }
+
+// MDMWindowsWipeType specifies what type of remote wipe we want
+// to perform.
+type MDMWindowsWipeType int
+
+const (
+	MDMWindowsWipeTypeDoWipe MDMWindowsWipeType = iota
+	MDMWindowsWipeTypeDoWipeProtected
+)
+
+var wipeTypeVariants = map[MDMWindowsWipeType]string{
+	MDMWindowsWipeTypeDoWipe:          "doWipe",
+	MDMWindowsWipeTypeDoWipeProtected: "doWipeProtected",
+}
+
+func (wt *MDMWindowsWipeType) String() string {
+	if wt == nil {
+		return "<nil>"
+	}
+	return wipeTypeVariants[*wt]
+}
+
+func (wt *MDMWindowsWipeType) UnmarshalJSON(b []byte) error {
+	var s string
+	if err := json.Unmarshal(b, &s); err != nil {
+		return err
+	}
+	for k, v := range wipeTypeVariants {
+		if v == s {
+			*wt = k
+			return nil
+		}
+	}
+	return fmt.Errorf("invalid WipeType: %s", s)
```

Comment:
:+1: Was wondering if the validation of the wipe type was missing when looking at the service method, that's a nice way to ensure it gets done!